cypress/e2e/article.cy.js

```diff
@@ -1,24 +1,121 @@
+    cy.get('h1').should('contain.text', articleTitle);
   });
 
-  it('should be edited using Edit button', () => {
+});
+
+describe(`User is able to edit an article`, () => {
+  let newTitle;
+  let newDescription;
+  let newBody;
+  let newTag;
 
+  beforeEach(() => {
+    newTitle = faker.lorem.sentence(2).slice(0, -1);
+    newDescription = faker.lorem.sentence(2).slice(0, -1);
+    newBody = faker.lorem.sentence(2).slice(0, -1);
+    newTag = faker.lorem.sentence(1).slice(0, -1);
+
+    cy.task('db:clear');
+    cy.registerNewUser().then(user => {
+      cy.login(user).then(() => {
+        cy.createNewArticle(user).then((slug) => {
+          cy.visit(`#/articles/${slug}`);
+          user.articleSlug = slug;
+        });
+      });
+    }).as('user');
   });
 
-  it('should be deleted using Delete button', () => {
+  it('User is able to edit the title of his own article.', () => {
+    cy.get('@user').then(user => {
+      cy.getByDataCy('edit-article-link').first().click();
+      cy.getByDataCy('article-title').clear().type(newTitle);
+      cy.getByDataCy('publish-btn').should('exist').click();
+      cy.get('h1').should('contain.text', newTitle)
+    });
+  });
 
+  it('User is able to edit the description of his own article.', () => {
+    cy.get('@user').then(user => {
+      cy.getByDataCy('edit-article-link').first().click();
+      cy.getByDataCy('article-description').clear().type(newDescription);
+      cy.getByDataCy('publish-btn').should('exist').click();
+      //cy.wait(2000);
+      cy.visit('#/');
+      cy.contains('p', `${newDescription}`).should('exist');
+    });
+  });
+
+  it('User is able to edit the body of his own article.', () => {
+    cy.get('@user').then(user => {
+      cy.getByDataCy('edit-article-link').first().click();
+      cy.getByDataCy('article-body').clear().type(newBody);
+      cy.getByDataCy('publish-btn').should('exist').click();
+      cy.get('p').should('contain.text', newBody);
+    });
+  });
+
+  it('User is able to edit the tag of his own article.', () => {
+    cy.get('@user').then(user => {
+      cy.getByDataCy('edit-article-link').first().click();
+      cy.getByDataCy('article-tags').first().type(`${newTag}{enter}`);
+      cy.getByDataCy('publish-btn').should('exist').click();
+      cy.contains('a', newTag).should('exist');
+    });
   });
 });
+
+describe(`User is able to delete an article`, () => {
+  beforeEach(() => {
+    cy.task('db:clear');
+    cy.registerNewUser().then(user => {
+      cy.login(user).then(() => {
+        cy.createNewArticle(user).then((slug) => {
+          cy.visit(`#/articles/${slug}`);
+          user.articleSlug = slug;
+        });
+      });
+    }).as('user');
+  });
+
+  it.only('User is able to delete an article by clicking on the Delete button', () => {
+    cy.get('@user').then(user => {
+      cy.getByDataCy('delete-article-btn').first().should('exist').click();
+      //cy.wait(2000);
+      cy.url('#/');
```

Comment:
The `cy.url` function should be used with `.should` like `cy.url().should('include', '/')`